ledger-tool/src/main.rs

```diff
@@ -2074,13 +2104,33 @@ fn main() {
                 );
                 exit(1);
             }
-            println!("Purging data from slots {} to {}", start_slot, end_slot);
-            if no_compaction {
-                blockstore.purge_slots(start_slot, end_slot, PurgeType::Exact);
-            } else {
-                blockstore.purge_and_compact_slots(start_slot, end_slot);
+            info!(
+                "Purging data from slots {} to {} ({} slots) (skip compaction: {})",
+                start_slot,
+                end_slot,
+                end_slot - start_slot,
+                no_compaction,
+            );
+            for slots in &(start_slot..=end_slot).chunks(batch_size) {
+                let slots = slots.collect::<Vec<_>>();
+                assert!(!slots.is_empty());
+
+                let start_slot = *slots.first().unwrap();
+                let end_slot = *slots.last().unwrap();
```

Comment:
I think this batching logic should be too simple to introduce bugs.

Reply:
And this should be semantically equivalent repeatedly executing `solana-ledger-tool purge ...` {`N1 N2`, `N2 N3`, `N3 N5 N4`}.